signer/src/block_observer.rs

```diff
@@ -288,6 +288,10 @@ where
                 let mut tx_bytes = Vec::new();
                 tx.consensus_encode(&mut tx_bytes)?;
 
+                // TODO: these aren't all sBTC transactions. Some of these
+                // could be "donations". One way to properly label these is
+                // to look at the scriptPubKey of the prevouts of the
+                // transaction.
```

Comment:
Yes, we know we can recognize a donation by it's `scriptPubKey` since they will be locked by something that is known. Specifically, donations are spend-path-only pay-to-taproot UTXOs where the locking public key is the signer aggregate key (the DKG aggregate key). We cannot recognize deposits at all, the signers need to be told that a transaction is a deposit (and then they verify whether that is true).